### PR TITLE
CI: Send CodeClimate test reports, deploy to RubyGems on tagging

### PR DESCRIPTION
**Pull request by iMacTia**
## Description
* Test Coverage Report is sent to CodeClimate
* Gem release happens on a separate stage

I'm not a Travis CI expert so appreciate your feedback on these changes 😄.
The idea was that I wanted the code climate script to be downloaded and executed only one, not for all `rvm`s.
Also, deployment is currently limited to a singular version thanks to the `on` option, but moving that into a stage allows to use the stage `rvm` option to do the same.

### CODE REVIEW

**Comment by iMacTia** [comment]
@technoweenie @olleolleolle I have to mark the PR as ready for review in order for Travis to build it 🤦‍♂️ 
Please do not merge it just yet

**Comment by olleolleolle** [comment]
I starting sketching on something like this, to use the simplecov output, for each Ruby version.

```yaml
after_script:
  - mkdir -p reports
  - export CURRENT_RUBY=`rvm-current | cut -c6-8` # "ruby-2.6.1" becomes "2.6"
  - tools/cc-test-reporter format-coverage -t simplecov -o reports/$CURRENT_RUBY.codeclimate.json # Format Ruby coverage
```

Inspiration: This CircleCI example shows the "format and save each step":
https://docs.codeclimate.com/docs/circle-ci-test-coverage-example#section-circleci-parallel-buildeac

**Comment by iMacTia** [comment]
@olleolleolle This is now working as expected, at least.
Up to you for a final review 😃. Let me know if there's anything else that might be improved

File: .travis.yml [hunk 1]
```diff
@@ -1,35 +1,51 @@
 language: ruby
-cache: bundler
+cache:
+  bundler: true
```
Comment (olleolleolle) [inline .travis.yml]: This can revert to the old, short-hand expression.

File: .travis.yml [hunk 2]
```diff
@@ -1,35 +1,51 @@
...
 jobs:
   include:
     - stage: linting
       rvm: ruby-2.4.1 # Pre-installed on TravisCI
       install: true # Do not bundle install
       script: gem install rubocop --no-document && rubocop
+    - stage: latest
+      rvm: 2.6 # Run custom before/after scripts for latest supported Ruby version.
+      before_script:
+        - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
```
Comment (olleolleolle) [inline .travis.yml]:
```suggestion
        - curl -o ./cc-test-reporter -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64
```

**Pull request merged**